Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceKeyboard.cpp

```diff
@@ -268,4 +327,33 @@ namespace AzFramework
             default: return nullptr;
         }
     }
+
+    AZStd::string XcbInputDeviceKeyboard::TextFromKeycode(xkb_state *state, xkb_keycode_t code)
+    {
+        // Find out how much of a buffer we need
+        const size_t size = xkb_state_key_get_utf8(state, code, nullptr, 0);
+        if (!size)
+        {
+            return {};
+        }
+        // xkb_state_key_get_utf8 will null-terminate the resulting string, and
+        // will truncate the result if there is not enough space for the null
+        // byte. The first call returns the size of the resulting string
+        // without including the null byte. Reserve enough space for the string
+        // + null byte, set the string's size, then use xkb_state_key_get_utf8
+        // to copy the string data
+        AZStd::string chars;
+        chars.reserve(size + 1);
+        chars.resize(size);
+        xkb_state_key_get_utf8(state, code, chars.data(), chars.size() + 1);
```

Comment:
`resize_no_construct` should be used to size the string to appropriate.

Furthermore the string class takes NUL-Termination into account internally, there is no need to reserve `size + 1`, plus the `resize/resize_no_construct` function will place a NUL-Terminator at the `size` index.

```suggestion
        AZStd::string chars;
        chars.resize_no_construct(size);
        xkb_state_key_get_utf8(state, code, chars.data(), chars.size());
```